### PR TITLE
User views cart with items userstory21 #55

### LINKED ISSUE #55: User Story 21, User views their cart show page with items in the cart
```
As a visitor or registered user
When I have added items to my cart
And I visit my cart ("/cart")
I see all items I've added to my cart
And I see a link to empty my cart
Each item in my cart shows the following information:
- the name of the item
- a very small thumbnail image of the item
- the merchant I'm buying this item from
- the price of the item
- my desired quantity of the item
- a subtotal (price multiplied by quantity)

I also see a grand total of what everything in my cart will cost
```

DO THESE STORIES FIRST:
- User Story 2, Visitor Navigation

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #57

child of #82

depends on #79

### PR DESCRIPTION
**Pull request by milevy1**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?
Added Cart index page for user story 21.  Includes two Cart instance methods and one Item instance method to push calculations to the model and clean up the view.  If there are a lot of questions, I can review this with the team tomorrow.

# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #55 

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:
My gift to you @blake-enyart 🎁

### CODE REVIEW

File: spec/features/cart/view_cart_with_items_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      click_on "Add to Cart"
+
+      visit item_path(item_2) # Add two item_2s to cart
+      click_on "Add to Cart"
+      visit item_path(item_2)
+      click_on "Add to Cart"
+
+      visit cart_path
+
+      within "#cart-item-#{item_1.id}" do
```
Comment (blake-enyart) [inline spec/features/cart/view_cart_with_items_spec.rb]: class id with -cards- for future for containers in styling later

File: spec/features/cart/view_cart_with_items_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        expect(page).to have_content(item_1.name)
+        expect(page).to have_css("img[src*='#{item_1.image_url}']")
+        expect(page).to have_content(item_1.user.name)
+        expect(page).to have_content("$#{item_1.price}")
+        expect(page).to have_content("Subtotal: #{item_1.price}")
```
Comment (milevy1) [inline spec/features/cart/view_cart_with_items_spec.rb]: Call subtotal method

File: app/views/cart/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+<% @cart.items.each do |item, quantity| %>
+  <section id="cart-item-<%= item.id %>">
+    <ul>
+      <li><%= link_to item.name, item_path(item) %></li>
+      <li><%= link_to image_tag(item.image_url, alt: "Image of #{item.name}"), item_path(item) %></li>
+      <li><%= item.user.name %></li>
+      <li>$<%= item.price.round(2) %></li>
+      <li>Quantity: <%= quantity %></li>
+      <li>Subtotal: <%= item.subtotal(quantity).round(2) %></li>
```
Comment (blake-enyart) [inline app/views/cart/index.html.erb]: Issue later to have 2 digits after the decimal if we use this method and the number ends in zero
